interview-questions.md

```diff
@@ -4,49 +4,60 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. Why would you use the method super()?
 
-  Your answer:
+  Your answer: super() is used to call upon the constructor() of the parent class. So inside a child class, it will have a super() to refer to all the properties of the parent class and you are able to generate more properties inside your child class.
 
-  Researched answer:
+  Researched answer: You would use the method super() to access and call functions from an object's parent. The method super() will make you child class inherit the methods from the parent class's constructor() method, therefor, the child has access to the parent's properties and methods.
 
 
 
 2. What is a virtual DOM?
 
-  Your answer:
+  Your answer: I think that a virtual DOM is possibly a representation of the actual DOM. So instead of actually manipulating a DOM directly, we manipulate the virtual DOM
 
-  Researched answer:
+  Researched answer: A virtual DOM object is a representation of a DOM object, just like a "lightweight" copy. A virtual DOM has the same properties as a real DOM but it lacks the real DOM's ability to change whats on the screen. Manipulating the virtual DOM is far faster because nothing is drawn onscreen. So the example given by codeacademy is like editing the blueprints of a house rather than moving actual rooms in an actual house. So React compares when an update happens: it takes the updated virtual DOMs with a copy of the virtual DOM before the update happened, and it responds back by changing only those objects with a different virtual DOM, this process is called "diffing" React:
 
+1- The entire virtual DOM gets updated.
+2- The virtual DOM gets compared to what it looked like before you updated it. React figures out which objects have changed.
+3- The changed objects, and the changed objects only, get updated on the real DOM.
+4- Changes on the real DOM cause the screen to change.
 
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: I know that they cannot obviously contain state, and I think that a basic class component is called a functional component. So you need to just build a regular function for the (props) section?
```

Comment:
Not quite. A functional component is different from a 'Class' component. A functional component is just that, a component that is created as a function. A class component, is one that starts with "class nameHere extends.."

Additionally, the question is asking about the parts necessary to create a class component, not what the difference between the two are.